### DISCUSSION TITLE
References to eclipse-platform-parent outside this repository

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-platform/eclipse.platform.releng.aggregator#1456] [Q&A] [answered]
There are some places outside this repository refering to the eclipse-platform-parent via a relative path `../eclipse-platform-parent` that can only work in _this_ repository: https://github.com/search?q=org%3Aeclipse-platform+eclipse-platform-parent&type=code&p=1 IMO. I noticed that when looking at platform-images. Are those just copy/paste errors and one could remove the relativePath tag from all other repos or am I missing some clue in my setup?

**Comment by HannesWell** [comment, marked as answer]
> There are some places outside this repository refering to the eclipse-platform-parent via a relative path `../eclipse-platform-parent` that can only work in _this_ repository: https://github.com/search?q=org%3Aeclipse-platform+eclipse-platform-parent&type=code&p=1 

AFAICT this is the case for all Eclipse TLPs, i.e. each sub-module repository of the `eclipse.platform.releng.aggregator`.

>I noticed that when looking at platform-images. Are those just copy/paste errors and one could remove the relativePath tag from all other repos or am I missing some clue in my setup?

This is intentional to use the local `eclipse-platform-parent` if those repos are checked out as sub-module of the releng.aggregator instead of the one in the eclipse-repo. But the maven build will also work if the target of the relative path does not exist, because then it searches the parent in usual repositories. The [documentation of Maven parent's `relativePath`](https://maven.apache.org/ref/3.9.5/maven-model/maven.html#class_parent) says the following:
```
The relative path of the parent pom.xml file within the check out. If not specified, it defaults to ../pom.xml.
Maven looks for the parent POM first in this location on the filesystem, then the local repository, and lastly in the remote repo.
relativePath allows you to select a different location, for example when your structure is flat, or deeper without an intermediate parent POM.
However, the group ID, artifact ID and version are still required, and must match the file in the location given
or it will revert to the repository for the POM. This feature is only for enhancing the development in a local checkout of that project.
Set the value to an empty string in case you want to disable the feature and always resolve the parent POM from the repositories.
```
